base/docs/basedocs.jl

```diff
@@ -2758,6 +2758,11 @@ The syntax `@atomic! max(a().b, c)` returns `modifyproperty!(a(), :b,
 max, c, :sequentially_consistent))`, where the first argument must be a
 `getfield` expression and is modified atomically.
 
+In the current implementation, the default implementation of `modifyproperty!` does not call
```

Comment:
Typically documentation is written in the imperative tense, and tries to avoid conditional modifiers of this sort.

Reply:
Any suggestions? How about something like

> Invocation of `op(getproperty(x, f), v)` must return a value that can be stored in the field `f` of the object `x`. In particular, unlike the default behavior of `setproperty!`, the `convert` function is not called automatically. However, this limitation may be removed in the future.

Reply:
Yep, that reads better. I think remove the comment about "in the future" though (we also can change errors -> working as a rule, so it is unnecessary to state it)

Reply:
> we also can change errors -> working as a rule

Yeah, it would greatly simply documentation like this. But some exceptions (e.g., `LinearAlgebra.SingularException`, `Base.IOError`) are used as an "output" of a function. So, I think we need to exclude some functions if we set this as a rule. Also, to handle APIs that want to guarantee to return an error with information in certain cases, I think we need something like `Union{Ok,Err}` convention #45080. After that, it'd be nice to define that changing throwing to not-throwing is non-breaking but changing return value, including `Union{Ok,Err}`, is breaking.